### PR TITLE
docs: initial defer docs

### PR DESCRIPTION
**Pull request by jacob-ebey**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

File: docs/components/await.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+function Accessor() {
+  const value = useAsyncValue();
+  return <p>{value}</p>;
+}
+// ...
+<Suspense>
+  <Await resolve={deferredValue}>
+    <Accessor />
+  </Await>
+</Suspense>;
```
Comment (machour) [inline docs/components/await.md:33]:
```suggestion
</Suspense>
```

File: docs/guides/streaming.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+      >
+        <Await
+          resolve={data.packageLocation}
+          errorElement={
+            <p>Error loading package location!</p>
+          }
+        >
+          <PackageLocation />
+        </Await>
+      </Suspense> //20
```
Comment (machour) [inline docs/guides/streaming.md:333]:
```suggestion
      </Suspense>
```

**Review by machour — APPROVED** [review]
:chiefkiss docs, well done!

File: docs/guides/streaming.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
+---
+title: Streaming
+description: When, why, and how to stream with React 18 and Remix's deferred API.
+---
+
```
Comment (jacob-ebey) [inline docs/guides/streaming.md:5]: Needs a title / h1

**Comment by kelvinsekx** [comment]
on line 7 of https://github.com/remix-run/remix/pull/5121/commits/2ea55482f8a8bab8b862067c74ddf0f0af11d174
`This can be thought of as a thin wrapper around React Error Boundaries with support for handling SSR that will suspend to resolves the data of a deferred loader value.`, there is a typo in this line. 

The **...suspend to resolves...** should have been written as **...suspend to resolve...**.

**Pull request merged**